### ISSUE TITLE
Warn *args/**kwargs using tuple/dict

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [PyCQA/flake8-pyi#273]
I've often seen some developers write an annotation like,

```python
def foo(*args: list[object], **kwargs: dict[str, object]):
  ...

```
when there intent was 

```python
def foo(*args: object, **kwargs: object):
  ...

```

Often with list instead of tuple[object, ...] as a lot of people first guess args are a list.

This rule is not always right, but I think it's rare for someone to actually want arbitrary keyword arguments that are dict. Spot checking internal codebase I see ~100 usages of this pattern (many are probably copy pastes as similar code) that all look like mistakes.

Does this rule sound reasonable to add?

**Comment by JelleZijlstra** [comment]
This is definitely a common pattern and it's useful to have some tool warn about it.

I'm not sure flake8-pyi is the right place, though, since it's meant for stubs and this issue is more likely to appear in implementation files. Maybe it could go into flake8-bugbear?

A similar issue is `tuple[int]`. That means a 1-tuple of an int, but 99% chance the author meant `tuple[int, ...]`.

**Comment by hmc-cs-mdrissi** [comment]
Closing this and reopening elsewhere sounds reasonable given stub focus. I do mostly see this in implementation files.

Is flake8-bugbear recommended place for .py type related lints?

**Comment by AlexWaygood** [comment]
> Is flake8-bugbear recommended place for .py type related lints?

I'm not really sure where the best place for this kind of thing is, to be honest. Somebody could/should probably make a plugin dedicated to runtime type-related lints of this kind. `flake8-bugbear` is _probably_ the best place for this kind of thing for now, though? :)

**Comment by AlexWaygood** [comment]
I'm closing this for now, since I think we're agreed that this check probably doesn't belong here, but good luck for finding a home for it! :-)
